boards: remove non-minimal peripherals from defconfig

gmarull opened the pull request:
According to the board porting guidelines, boards should "leave
peripherals and their drivers disabled by default". In Zephyr we
tipically enable GPIO and SERIAL, as they are virtually required by all
samples/tests in tree. However, for the rest of peripherals it is up to
the application/test to enable the necessary driver classes. It is also
useful that board's Kconfig.defconfig enables certain driver peripherals
based on a condition, e.g. enable I2C if SENSOR=y.

Ref. https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#general-recommendations

This patch removes the following driver classes from defconfig files:

- CONFIG_ADC
- CONFIG_COUNTER
- CONFIG_EEPROM
- CONFIG_ENTROPY
- CONFIG_ESPI
- CONFIG_HWINFO
- CONFIG_I2C
- CONFIG_LED
- CONFIG_NETWORKING
- CONFIG_PS2
- CONFIG_PWM
- CONFIG_SENSOR
- CONFIG_SPI
- CONFIG_SPI_SLAVE
- CONFIG_WATCHDOG

Note that a previous attempt was done in #38510.

Fixes #30694

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: remove non-minimal peripherals from defconfig

According to the board porting guidelines, boards should "leave
peripherals and their drivers disabled by default". In Zephyr we
tipically enable GPIO and SERIAL, as they are virtually required by all
samples/tests in tree. However, for the rest of peripherals it is up to
the application/test to enable the necessary driver classes. It is also
useful that board's Kconfig.defconfig enables certain driver peripherals
based on a condition, e.g. enable I2C if SENSOR=y.

Ref. https://docs.zephyrproject.org/latest/hardware/porting/
board_porting.html#general-recommendations

This patch removes the following driver classes from defconfig files:

- CONFIG_ADC
- CONFIG_COUNTER
- CONFIG_EEPROM
- CONFIG_ENTROPY
- CONFIG_ESPI
- CONFIG_HWINFO
- CONFIG_I2C
- CONFIG_LED
- CONFIG_NETWORKING
- CONFIG_PS2
- CONFIG_PWM
- CONFIG_SENSOR
- CONFIG_SPI
- CONFIG_SPI_SLAVE
- CONFIG_WATCHDOG

Note that a previous attempt was done in #38510.

Fixes #30694

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

nashif commented:
marked DNM due to an issue with CI not calculating correct size of matrix, need a fix to testplan first.

nashif commented:
need to get this in first: https://github.com/zephyrproject-rtos/zephyr/pull/48576

nashif commented:
@gmarull  can you rebase please.

gmarull committed:
samples: drivers: lcd_cyclonev_socdk: add missing CONFIG_I2C=y

The sample relies on using I2C API, so it must enable CONFIG_I2C.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: ip_k66f: enable SPI if NET_DSA=y

SPI is required by DSA_KSZ8794, enabled if NET_SDA=y.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
samples: modules: tflite-micro: magic_wand: add missing CONFIG_I2C=y

The magic_want sample requires I2C bus to be enabled (used sensor is
connected to the bus).

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

evgeniy-paltsev reviewed (commented):
Hmm, what about peripheral chains? For example we have GPIO expander which is connected via I2C. Obviously the samples which uses GPIO (e.g. samples/basic/blinky) doesn't know anything about it and it enables only `CONFIG_GPIO`. But now we don't enable I2C in board's defconfig. So, where it should be enabled?

gmarull commented:
> Hmm, what about peripheral chains? For example we have GPIO expander which is connected via I2C. Obviously the samples which uses GPIO (e.g. samples/basic/blinky) doesn't know anything about it and it enables only `CONFIG_GPIO`. But now we don't enable I2C in board's defconfig. So, where it should be enabled?

For such cases you can do in `Kconfig.defconfig`:

```
config I2C
	default y if GPIO
```

galak commented on boards/arm/mec1501modular_assy6885/mec1501modular_assy6885_defconfig:
```diff
@@ -17,13 +17,5 @@ CONFIG_UART_CONSOLE=y
-CONFIG_I2C=y
-CONFIG_I2C_INIT_PRIORITY=60
-CONFIG_ESPI=y
-CONFIG_ESPI_PERIPHERAL_UART_SOC_MAPPING=1
```
are you sure you can remove `CONFIG_ESPI_PERIPHERAL_UART_SOC_MAPPING`?

gmarull replied:
added to defconfig, thanks

galak commented on boards/riscv/litex_vexriscv/litex_vexriscv_defconfig:
```diff
@@ -11,19 +11,7 @@ CONFIG_LITEX_TIMER=y
 CONFIG_UART_CONSOLE=y
-CONFIG_NETWORKING=y
-CONFIG_ETH_LITEETH=y
-CONFIG_ETH_LITEETH_0=y
```
not sure you can remove CONFIG_ETH_LITEETH_0/CONFIG_ETH_LITEETH (may need to move to Kconfig.defconfig

gmarull replied:
As it is done often, it's the application/sample that is responsible for that when SoC doesn't select such defaults in SoC Kconfig.defconfig files. This should be properly fixed by using the `DT_HAS*` helpers.

galak replied:
Agreed this should move to DT_HAS_*, but for now is this not breaking anything?

gmarull replied:
Moved options to `Kconfig.defconfig`, please check latest changeset. Done at board level (not SoC level) as not all boards may want such behavior if NETWORKING=y. I've seen quite a few boards doing similar thing for the L2 option.

galak commented on boards/arm/mec15xxevb_assy6853/mec15xxevb_assy6853_defconfig:
```diff
@@ -15,13 +15,6 @@ CONFIG_UART_CONSOLE=y
-CONFIG_PWM=y
-CONFIG_SENSOR=y
-CONFIG_I2C=y
-CONFIG_I2C_INIT_PRIORITY=60
```
I suggest putting removal of CONFIG_I2C_INIT_PRIORITY in a different PR.

gmarull replied:
added specific commit, please check

galak reviewed (changes requested):
Marking 'request changes' to address a few questions and comments.

gmarull committed:
boards: mec1501modular_assy6885: move UART_SOC_MAPPING to Kconfig.defconfig

Set the ESPI_PERIPHERAL_UART_SOC_MAPPING default value to 1 in
Kconfig.defconfig, only if ESPI=y.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: mec15xxevb_assy6853: remove redundant I2C_INIT_PRIORITY

In this board I2C is required to talk to a GPIO expander (PCA95XX). By
default I2C devices are initialized with KERNEL_INIT_PRIORITY_DEVICE
(set to 50), and the PCA95XX driver is initialized with priority 70.
This means that there is no need to customize I2C priority as it will
always come before.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: mec1501modular_assy6885: remove redundant I2C_INIT_PRIORITY

This board has no devices connected to the I2C bus, so there's no need
to customize init priorities at the board level.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
drivers: counter: increase COUNTER_INIT_PRIORITY

Some counter devices depend on buses (e.g. I2C for Maxim DS3231). Bus
devices are tipically initialized with KERNEL_INIT_PRIORITY_DEVICE as
well, so there's no guarantee counter will be initialized be initialized
before without manually tweaking priorities.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull committed:
boards: cyclonev_socdk: remove redundant COUNTER_INIT_PRIORITY

Counter devices are now initialized with priority 60 by default, which
guarantees it will come after I2C (50).

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

gmarull commented on boards/arm/mec1501modular_assy6885/mec1501modular_assy6885_defconfig:
```diff
@@ -17,13 +17,5 @@ CONFIG_UART_CONSOLE=y
-CONFIG_PWM=y
-CONFIG_SENSOR=y
-CONFIG_I2C=y
-CONFIG_I2C_INIT_PRIORITY=60
```
added specific commit, please check

nashif commented:
Would like to see most of not all "owners" ACK this and approve before it is merged. I agree with the changes, but we need to give time for this to be reviewed and any dependencies adjusted.

nandojve commented on boards/arc/hsdk/hsdk_2cores_defconfig:
```diff
@@ -12,6 +12,5 @@ CONFIG_CONSOLE=y
 CONFIG_UART_CONSOLE=y
 CONFIG_SERIAL=y
 CONFIG_GPIO=y
```
I`m in favor to drop GPIO too. It became so essential that all boards must have it. In this case, I think it will be better define in the SoC with default=y. Intention is mimics same use case that some pinctrl already have been using in the project. Like this https://github.com/zephyrproject-rtos/zephyr/blob/8e55e59c5917b61d563d5e8bf759e8aeb746ff8b/soc/arm/gigadevice/Kconfig.defconfig#L8-L9

gmarull replied:
that's a good point. I wonder if we should go this route for other things as well (e.g. MPU). Let's discuss this in architecture meeting @carlescufi.

fkokosinski reviewed (dismissed):
Hi, I've tested your changes for these boards:
* `hifive1`
* `hifive1_revb`
* `litex_vexriscv`

chrta reviewed (dismissed):
Tested for efr32mg_sltb004

gmarull commented:
rebased to fix conflicts

chrta reviewed (dismissed):
ACK for efr32mg_sltb004a

gmarull committed:
boards: litex_vexriscv: enable ETH_LITEETH(0) and L2 if NETWORKING

Automatically enable:

CONFIG_ETH_LITEETH
CONFIG_ETH_LITEETH0
CONFIG_NET_L2_ETHERNET

if CONFIG_NETWORKING=y.

Signed-off-by: Gerard Marull-Paretas <gerard.marull@nordicsemi.no>

jvasanth1 commented:
mec15xx and mec1501 updates look good.

gmarull commented:
@chrta @fkokosinski @anangl @glaubermaroto @stephanosio please re-review
